### DISCUSSION TITLE
Is it possible to have a virtualenvwrapper formula?

### DISCUSSION THREAD

**Discussion by georgexsh** [Homebrew/discussions#1083] [Everyday usage]
Currently, I am using `pipx` to install `virtualenvwrapper`, but it is very painful if python patch upgrades, I have to reinstall-all packages, if `virtualenvwrapper` is managed by homebrew, that will be much easier.
Is there any particular reason that there is not a `virtualenvwrapper` formula, it is odd because IMO it is a commonly used python tool.

**Comment by carlocab** [comment]
It might be possible. The only way to find out is to open a PR so it can be reviewed. Here are some docs to help you get started:

https://docs.brew.sh/Formula-Cookbook
https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request
https://docs.brew.sh/Python-for-Formula-Authors

You may also wish to have a look at other formulae that depend on Python for ideas on how to write the formula.

**Reply by georgexsh** [reply]
thanks for the guides, my previous concern is homebrew use virtualenv to isolate python packages, virtualenv formula may incompatible, then I have found `virtualenv_install_with_resources` uses `python -m venv`, so this should be alright.

**Reply by georgexsh** [reply]
PR created for virtualenv first https://github.com/Homebrew/homebrew-core/pull/73531

**Reply by georgexsh** [reply]
PR for virtualenvwrapper https://github.com/Homebrew/homebrew-core/pull/75681
